app/src/main/java/org/oppia/app/player/state/itemviewmodel/FractionInteractionViewModel.kt

```diff
@@ -1,32 +1,89 @@
 package org.oppia.app.player.state.itemviewmodel
 
 import android.content.Context
+import android.text.Editable
+import android.text.TextWatcher
+import androidx.databinding.Bindable
+import androidx.databinding.ObservableField
 import org.oppia.app.R
 import org.oppia.app.model.Interaction
 import org.oppia.app.model.InteractionObject
 import org.oppia.app.model.UserAnswer
 import org.oppia.app.parser.StringToFractionParser
+import org.oppia.app.player.state.StateFragment
+import org.oppia.app.player.state.answerhandling.AnswerErrorCategory
 import org.oppia.app.player.state.answerhandling.InteractionAnswerHandler
 
 /** [ViewModel] for the fraction input interaction. */
 class FractionInteractionViewModel(
-  interaction: Interaction, private val context: Context
+  interaction: Interaction, private val context: Context, private val interactionAnswerHandler: InteractionAnswerHandler
 ) : StateItemViewModel(ViewType.FRACTION_INPUT_INTERACTION), InteractionAnswerHandler {
+  private var pendingAnswerError: String? = null
   var answerText: CharSequence = ""
+  var errorMessage = ObservableField<String>("")
   val hintText: CharSequence = deriveHintText(interaction)
+  val stringToFractionParser: StringToFractionParser
+
+  init {
+    stringToFractionParser = StringToFractionParser()
+  }
 
   override fun getPendingAnswer(): UserAnswer {
     val userAnswerBuilder = UserAnswer.newBuilder()
     if (answerText.isNotEmpty()) {
       val answerTextString = answerText.toString()
       userAnswerBuilder.answer = InteractionObject.newBuilder()
-        .setFraction(StringToFractionParser().getFractionFromString(answerTextString))
+        .setFraction(stringToFractionParser.getFractionFromString(answerTextString))
         .build()
       userAnswerBuilder.plainAnswer = answerTextString
     }
     return userAnswerBuilder.build()
   }
 
+  override fun hasPendingAnswerErrorOnSubmit(): Boolean {
+    setPendingAnswerError(AnswerErrorCategory.SUBMIT_TIME)
+    return getPendingAnswerError() != null
+  }
+
+  /**
+   * It set pendingAnswerError, send [errorMessage] and pendingAnswerError to the presenter via [StateFragment]
+   */
+  fun setPendingAnswerError(category: AnswerErrorCategory) {
```

Comment:
Maybe ``checkPendingAnswerError`` is a bit clearer, since this is more than just a trivial setter--it needs to perform a lookup to get the answer error.

Reply:
renamed to checkPendingAnswerError